mesop/server/server.py

```diff
@@ -125,7 +128,14 @@ def generate_data(ui_request: pb.UiRequest) -> Generator[str, None, None]:
       elif ui_request.HasField("user_event"):
         event = ui_request.user_event
         runtime().context().set_viewport_size(event.viewport_size)
-        runtime().context().update_state(event.states)
+
+        if event.states.states:
+          runtime().context().update_state(event.states)
+        else:
```

Comment:
Ah good point about that scenario for the memory case. It wouldn't recover correctly, so we'd still need to generate a some kind of unique token along with the uuid. Which in that case, maybe we can just rely on generating the random token each time.

--------

I think the reason for putting the UUID in the Flask Session is mainly to avoid tampering. If the UUID were passed in with the request, a user could use it from another client. However I guess it would still be possible to do this with the session cookie as well. But I think it would be harder for someone to get the session cookie than getting access to the UUID from the response?

But I think that's moot since we do need to generate a unique token each time or calculate the state hash.

Reply:
So I think what I'll do is just remove secret key part and rely solely on regenerating the token after each request. And I think we can prepend a timestamp to the token to ensure a bit more uniqueness.

Reply:
Ok. Removed the Flask session stuff for now.